### PR TITLE
website: support hidden pages in nav-data

### PR DESCRIPTION
**Pull request by zchsh**
👀 [Preview](https://nomad-git-zssupport-hidden-pages-hashicorp.vercel.app)
🎟️ [Asana task](https://app.asana.com/0/1100423001970639/1200197752405255/f)

This PR is a follow-up to #10425, and implements a more permanent and clean solution for "hidden" pages, via updates to our `docs-page` component.

---

Items in `nav-data.json` now support an optional `"hidden": true` property, which will hide the item (and any descendants) from the docs sidebar.

Alongside this change, all `.mdx` files in each `content` folder must be included in the corresponding `nav-data` file. The changes to `docs-page` included an automated check which will throw an error during the website build process if there are any unlinked content files.

### CODE REVIEW

File: .github/workflows/check-unlinked-content.js [hunk 1]
```diff
@@ -1,134 +0,0 @@
-var fs = require("fs");
```
Comment (zchsh) [inline .github/workflows/check-unlinked-content.js:1]: This check is no longer needed, it has been implemented in our `docs-page` component as part of [react-components#220](https://github.com/hashicorp/react-components/pull/220).

File: website/package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "@hashicorp/react-button": "5.0.1",
     "@hashicorp/react-call-to-action": "3.0.1",
     "@hashicorp/react-content": "7.0.1",
-    "@hashicorp/react-docs-page": "13.1.1",
+    "@hashicorp/react-docs-page": "13.1.7-alpha.10",
```
Comment (zchsh) [inline website/package.json:17]: Will bump this to a stable release once these PRs are finalized and https://github.com/hashicorp/react-components/pull/220 is merged and published.

**Review by jescalan — APPROVED** [review]

File: website/package.json (hunk 2 above)
Comment (zchsh) [reply website/package.json]: ✅ Bumped to stable

**Review by jescalan — APPROVED** [review]

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**